### ISSUE TITLE
[17.0] Product Category: When you save the settings, the Inventory Valuation method change to Manual 

### ISSUE THREAD

**Issue by MR-ILYAS08** [odoo/odoo#171353]
### Odoo Version

- [ ] 15.0
- [ ] 16.0
- [X] 17.0
- [ ] Other (specify)

### Steps to Reproduce

- Chose any Product Category
- Change 'Inventory Valuation' method  to Automated and save the change
- Go to settings module and click 'Save'
- Check again the product Category

The valuation method should be changed to manual

**Comment by hyz1187** [comment]
def set_values(self):
        automatic_before = self.env.user.has_group('stock_account.group_stock_accounting_automatic')
        super().set_values()
        if automatic_before and not self.group_stock_accounting_automatic:
            self.env['product.category'].sudo().with_context(active_test=False).search([
                ('property_valuation', '=', 'real_time')]).property_valuation = 'manual_periodic'
     I don't know why the value of this' self. group_stock_countingautomatic 'method is false?

**Comment by MR-ILYAS08** [comment]
Yes i had to show the box in setting and activate so i can work on it, but definitely shouldn't be the case
